tests/main/interfaces-desktop-launch/task.yaml

```diff
@@ -0,0 +1,56 @@
+summary: Ensure that the desktop-launch interface works.
+
+details: |
+    The desktop-launch interface allows a snap to launch other snaps via
+    the desktop files they provide to the host system.
+
+systems: [-ubuntu-core-*]
+
+prepare: |
+    if ! tests.session has-session-systemd-and-dbus; then
+        exit 0
+    fi
+    tests.session -u test prepare
+    tests.session -u test exec systemctl --user \
+      set-environment XDG_DATA_DIRS=/usr/share:/var/lib/snapd/desktop
+
+restore: |
+    if ! tests.session has-session-systemd-and-dbus; then
+        exit 0
+    fi
+    tests.session -u test restore
+    rm -f ~test/snap/test-app/current/launch-data.txt
+
+execute: |
+    if ! tests.session has-session-systemd-and-dbus; then
+        exit 0
+    fi
+
+    #shellcheck source=tests/lib/snaps.sh
+    . "$TESTSLIB"/snaps.sh
+
+    echo "Install the application snap"
+    install_local test-app
+    echo "The snap installs a desktop file"
+    [ -f /var/lib/snapd/desktop/applications/test-app_test-app.desktop ]
+
+    echo "Install the launcher snap"
+    install_local test-launcher
+
+    echo "The desktop-launch plug is initially disconnected"
+    snap connections test-launcher | MATCH "desktop-launch +test-launcher:desktop-launch +- +-"
+
+    echo "The plug can be connected"
+    snap connect test-launcher:desktop-launch
+    snap connections test-launcher | MATCH "desktop-launch +test-launcher:desktop-launch +:desktop-launch +manual"
+
+    echo "The launcher snap can launch other snaps via userd"
+    tests.session -u test exec test-launcher \
```

Comment:
does this need to be `exec`? I assume this is somehow related to user sessions?

Reply:
also should we be checking the output of this command with MATCH, since we are using `--print-reply`?